### PR TITLE
Alter log format to provide more numeric data

### PR DESCRIPTION
**Pull request by rjackson90**
The current JSON log format puts double quotes around the value of most fields, which causes downstream parsers to interpret the value as a string. For certain fields, we would like the data to be interpreted as numbers to aid downstream analysis. Removing the double quotes encapsulating the values is an attempt to signal to downstream parsers that the field value is numeric.

### CODE REVIEW

File: nginx.conf [hunk 1]
```diff
@@ -31,17 +31,17 @@ http {
         '"request_uri":"$request_uri",'
         '"request_length":"$request_length",'
         '"status":"$status",'
```
Comment (patrickf55places) [inline nginx.conf:33]: Should status also be treated as an int?

**Review by patrickf55places — APPROVED** [review]
Are there going to be any log ingestion issues as a result of the string->int change?

**Comment by patrickf55places** [comment]
Is it possible for any of these values to be missing (either an empty string, or null) that can cause JSON parsing errors?

**Comment by rjackson90** [comment]
@patrickf55places you raise some excellent questions, and unfortunately I don't have answers. I'm going to work with @alexberryman to use this version of the base docker image to test out the impact with a particular service. I don't think I've seen these not be available, so I'm not _super_ worried about that, especially since this is all information which originates within nginx and doesn't come from clients. I don't have strong feelings about making the HTTP code a number, but I feel like it's more like an enum than an actual number, so a string representation makes more sense. I'm open to other ideas though.

**Comment by patrickf55places** [comment]
Re: status code as a number, it is often useful to do range testing on status codes (e.g. `status >= 500` or `status >= 400 && status <= 499`, which becomes difficult to do with strings (though still possible if you can do pattern matching, e.g. `status ~= /^4/`

**Comment by alexberryman** [comment]
Looking at LS HTTP logs, there are instances of the upstream times being `-` when the there is an upstream timeout: [kibana document](https://logs.neighborhoods.com/app/kibana#/doc/3dc503a0-9853-11e9-b9a0-19ee76119d67/logstash-2019.08.21/fluentd?id=rIXRtGwBxWb9-ennX3Ig&_g=(refreshInterval:(display:Off,pause:!f,value:0),time:(from:now-4h,mode:quick,to:now)))

```
{
  "_index": "logstash-2019.08.21",
  "_type": "fluentd",
  "_id": "rIXRtGwBxWb9-ennX3Ig",
  "_score": 1,
  "_source": {
    "stream": "stdout",
    "docker": {
      "container_id": "d0fcc268ffe0fd09731aa9a291e7854a04bc160e23d41d9a84ae652a1cbe8fd8"
    },
    "kubernetes": {
      "container_name": "log-nginx-access",
      "namespace_name": "listing-service",
      "pod_name": "listing-service-http-b69dcbb68-twqgz",
      "container_image": "neighborhoods/logwatcher:1.0.0",
      "container_image_id": "docker-pullable://neighborhoods/logwatcher@sha256:ea6ddafb11f3a2199692acae0a03436dbc8f9d688250be28700d085b28174b07",
      "pod_id": "a00c82f0-c38f-11e9-abfb-0e081f397254",
      "host": "ip-10-128-4-132.ec2.internal",
      "labels": {
        "app": "listing-service",
        "newrelic": "enabled",
        "release": "listing-service",
        "tier": "http"
      }
    },
    "container_info": "d0fcc268ffe0fd09731aa9a291e7854a04bc160e23d41d9a84ae652a1cbe8fd8-stdout",
    "meta_tag": "kube.listing-service.listing-service-http-b69dcbb68-twqgz.log-nginx-access._labels.http",
    "client": "172.16.40.81",
    "scheme": "http",
    "method": "GET",
    "request_uri": "/HTTP1/PropertyCoordinates/?searchCriteria%5Bfilters%5D%5B0%5D%5Bglue%5D=and&searchCriteria%5Bfilters%5D%5B0%5D%5Bfield%5D=coordinates&searchCriteria%5Bfilters%5D%5B0%5D%5Bcondition%5D=st_within&searchCriteria%5Bfilters%5D%5B0%5D%5Bvalues%5D%5B0%5D=POLYGON%28%28-180+73.55567595831731%2C+-180+-89.99999893948277%2C+180+-89.99999893948277%2C+180+73.55567595831731%2C+-180+73.55567595831731%29%29&searchCriteria%5Bfilters%5D%5B1%5D%5Bglue%5D=and&searchCriteria%5Bfilters%5D%5B1%5D%5Bfield%5D=listing_status&searchCriteria%5Bfilters%5D%5B1%5D%5Bcondition%5D=in&searchCriteria%5Bfilters%5D%5B1%5D%5Bvalues%5D%5B0%5D=active&searchCriteria%5Bfilters%5D%5B2%5D%5Bglue%5D=and&searchCriteria%5Bfilters%5D%5B2%5D%5Bfield%5D=num_bedrooms&searchCriteria%5Bfilters%5D%5B2%5D%5Bcondition%5D=gte&searchCriteria%5Bfilters%5D%5B2%5D%5Bvalues%5D%5B0%5D=5&searchCriteria%5Bfilters%5D%5B3%5D%5Bglue%5D=and&searchCriteria%5Bfilters%5D%5B3%5D%5Bfield%5D=current_price&searchCriteria%5Bfilters%5D%5B3%5D%5Bcondition%5D=gte&searchCriteria%5Bfilters%5D%5B3%5D%5Bvalues%5D%5B0%5D=250000&searchCriteria%5Bfilters%5D%5B4%5D%5Bglue%5D=and&searchCriteria%5Bfilters%5D%5B4%5D%5Bfield%5D=current_price&searchCriteria%5Bfilters%5D%5B4%5D%5Bcondition%5D=lte&searchCriteria%5Bfilters%5D%5B4%5D%5Bvalues%5D%5B0%5D=350000&searchCriteria%5BpageSize%5D=200&searchCriteria%5BsortOrder%5D%5B0%5D%5Bfield%5D=imported_at&searchCriteria%5BsortOrder%5D%5B0%5D%5Bdirection%5D=DESC",
    "request_length": "3466",
    "status": "499",
    "bytes_sent": "0",
    "body_bytes_sent": "0",
    "referer": "",
    "user_agent": "Phalcon HTTP/0.0.2 (Curl)",
    "upstream_addr": "127.0.0.1:9000",
    "upstream_status": "-",
    "request_time": "29.999",
    "upstream_response_time": "-",
    "upstream_connect_time": "-",
    "upstream_header_time": "-",
    "upstream_bytes_received": "0",
    "meta_config_source": "nginx_access_perf label",
    "meta_status": "processed",
    "@timestamp": "2019-08-21T15:35:13.000000000+00:00"
  },
  "fields": {
    "@timestamp": [
      "2019-08-21T15:35:13.000Z"
    ]
  }
}
```

We should see how these timeouts behave with this new numerical format.

**Comment by alexberryman** [comment]
NGINX 499s seem to be where this is commonly happening https://httpstatuses.com/499

**Comment by rjackson90** [comment]
ugh, these changes will break everything then. Good find @alexberryman, but it looks like we'll have to parse these times out in fluentd.

**Pull request closed without merging**